### PR TITLE
Resolved displaying customer full name on admin order detail page

### PR DESCRIPTION
**Pull request by kazim-krish**
### Description (*)
#23627 : Fixed displaying customer full name on admin order detail page.

### Fixed Issues (if relevant)
1. #23627 : Fixed displaying customer full name on admin order detail page.

### Manual testing scenarios (*)
1. Registered a new customer will prefix, first name, middle name, last name, and suffix.
2. Placed an order with registered customer and noted that full customer name along with prefix and suffix is rendered on admin order detail page.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by orlangur — CHANGES_REQUESTED** [review]
@kazim-krish please get rid of unnecessary spaces in case some of methods return empty string.

Can we reuse some existing logic or create a new class encapsulating it also?

**Comment by gwharton** [comment]
Shouldn't any proposed solution respect the Magento Settings 

```
Stores -> Configuration -> Customers -> Customer Configuration -> Name and Address Options -> Show Prefix
Stores -> Configuration -> Customers -> Customer Configuration -> Name and Address Options -> Show Middle Name
Stores -> Configuration -> Customers -> Customer Configuration -> Name and Address Options -> Show Suffix
```

**Comment by sidolov** [comment]
@kazim-krish , please, see comment from @gwharton

**Comment by kazim-krish** [comment]
@sidolov @gwharton: How about if Order has customer prefix or customer middle name or customer suffix but Magento Customer Configuration -> Show Prefix or -> Show Middle Name or -> Show Suffix is disabled. If we keep with respect to Magento Customer configuration on admin order detail page, it would not show prefix or middle name or suffix if set off in customer configuration.
Case: Firstly Magento customer configuration to show prefix and middle name are on, the customer has already placed many orders from the frontend (that have a prefix and middle name in address). Later admin user disabled Magento customer configuration to show prefix and middle name. This will restrict current frontend customer to enter a prefix and middle name and also it will not show prefix and middle name to the earlier order placed on the admin order detail page.

**Comment by sidolov** [comment]
@kazim-krish it's expected behavior if you disable something the user hasn't the ability to use that and it does not matter if such info were collected before.

File: app/code/Magento/Sales/Model/Order.php [hunk 1]
```diff
@@ -1968,13 +1977,63 @@ public function getRelatedObjects()
...
         }
         return $customerName;
     }
 
+    /**
+     * Return Customer Address Prefixshow availability.
+     *
+     * @return bool|string
+     */
+    private function isAllowShowPrefix()
```
Comment (sidolov) [inline app/code/Magento/Sales/Model/Order.php:2000]: Please, add strict types

File: app/code/Magento/Sales/Model/Order.php [hunk 2]
```diff
@@ -1968,13 +1977,63 @@ public function getRelatedObjects()
...
+        );
+        return $isAllowed;
+    }
+
+    /**
+     * Return Customer Address Middlename availability.
+     *
+     * @return bool
+     */
+    private function isAllowShowMiddlename()
```
Comment (sidolov) [inline app/code/Magento/Sales/Model/Order.php:2014]: Please, add strict types

File: app/code/Magento/Sales/Model/Order.php [hunk 3]
```diff
@@ -1968,13 +1977,63 @@ public function getRelatedObjects()
...
+        );
+        return $isAllowed;
+    }
+
+    /**
+     * Return Customer Address Prefixshow availability.
+     *
+     * @return bool|string
+     */
+    private function isAllowShowSuffix()
```
Comment (sidolov) [inline app/code/Magento/Sales/Model/Order.php:2028]: Please, add strict types

File: app/code/Magento/Sales/Model/Order.php [hunk 4]
```diff
@@ -2750,7 +2809,11 @@ public function getCustomerLastname()
      */
     public function getCustomerMiddlename()
     {
-        return $this->getData(OrderInterface::CUSTOMER_MIDDLENAME);
+        $isAllowed = $this->isAllowShowMiddlename();
+        if ($isAllowed) {
+            return $this->getData(OrderInterface::CUSTOMER_MIDDLENAME);
+        }
+        return false;
```
Comment (sidolov) [inline app/code/Magento/Sales/Model/Order.php:2816]: Please, return null instead of false

File: app/code/Magento/Sales/Model/Order.php [hunk 5]
```diff
@@ -2780,7 +2843,11 @@ public function getCustomerNoteNotify()
      */
     public function getCustomerPrefix()
     {
-        return $this->getData(OrderInterface::CUSTOMER_PREFIX);
+        $isAllowed = $this->isAllowShowPrefix();
+        if (!empty($isAllowed)) {
+            return $this->getData(OrderInterface::CUSTOMER_PREFIX);
+        }
+        return false;
```
Comment (sidolov) [inline app/code/Magento/Sales/Model/Order.php:2850]: Please, return null instead of false

File: app/code/Magento/Sales/Model/Order.php [hunk 6]
```diff
@@ -2790,7 +2857,11 @@ public function getCustomerPrefix()
      */
     public function getCustomerSuffix()
     {
-        return $this->getData(OrderInterface::CUSTOMER_SUFFIX);
+        $isAllowed = $this->isAllowShowSuffix();
+        if (!empty($isAllowed)) {
+            return $this->getData(OrderInterface::CUSTOMER_SUFFIX);
+        }
+        return false;
```
Comment (sidolov) [inline app/code/Magento/Sales/Model/Order.php:2864]: Please, return null instead of false

**Review by sidolov — CHANGES_REQUESTED** [review]

**Comment by sidolov** [comment]
@kazim-krish I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**